Automattic/wp-calypso
# Tracking: fix inserter menu no results issue

#### Changes proposed in this Pull Request

This PR changes the way to track `no-results` search result from the `Inserter Menu`, replacing event delegation by using an [Slot](https://developer.wordpress.org/block-editor/developers/slotfills/), through of the, experimental, inserter menu extension.

Also, it implements a fallback implementation for the `7.8.1` core plugin version, since unfortunately, it has a bug. Basically, the `hasItems` parameter is always `true`, so we can't rely on it.
So, it checks the plugin version and applies the respective behavior according to this value. We should remove the fallback code once a new version of core-editor ships.

#### Markable changes

* Remove the event delegation tracker.
* Track same events using the `__experimentalInserterMenuExtension` component.
* It applies a temporary hacky fix to detect no-results search 

#### Testing instructions

1) Update your testing site with these changes (SB)
2) In your testing site, open the client dev console and set up the debug

```
> localStorage.setItem( 'debug', 'wpcom-block-editor*' )
```

A hard refresh is needed.

3) Open the Inserter Menu
4) Open the `Inserter Menu` selector
5) Generate event typing there
6) Confirm that it behaves as expected. It means when you get results, it should trigger a `wpcom_block_picker_search_term` event, as the screenshot shows below.

**blocks found** event
<img width="1463" alt="Screen Shot 2020-04-15 at 10 13 04 PM" src="https://user-images.githubusercontent.com/77539/79403892-8e5be400-7f66-11ea-95f5-790706ff3f87.png">

When there aren't blocks there, it also should trigger the `wpcom_block_picker_no_results` event too:

**blocks not found** event
<img width="1221" alt="Screen Shot 2020-04-15 at 10 13 58 PM" src="https://user-images.githubusercontent.com/77539/79403878-87cd6c80-7f66-11ea-8e51-64de9b20618f.png">

You can see what version is currently running in your site inspecting the debug line:

<img width="675" alt="Screen Shot 2020-04-16 at 8 15 16 AM" src="https://user-images.githubusercontent.com/77539/79449915-744bf100-7fba-11ea-9d6a-cf3a2cb9aa2b.png">


Fixes https://github.com/Automattic/wp-calypso/issues/41149
